src/gql/query/staking.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  DeepPartial,
+  GQLDelegation,
+  GQLQuery,
+  GQLQueryGqlRedelegationsArgs,
+  GQLQueryGqlUnbondingsArgs,
+  GQLRedelegation,
+  GQLStakingGqlDelegationsArgs,
+  GQLStakingGqlHistoryArgs,
+  GQLStakingGqlValidatorsArgs,
+  GQLStakingHistoryItem,
+  GQLUnbonding,
+  GQLValidator,
+  convertObjectToPropertiesString,
+  doGqlQuery,
+  gqlQuery,
+} from "../utils"
+
+export type QueryStakingArgs = {
+  delegations?: GQLStakingGqlDelegationsArgs
+  history?: GQLStakingGqlHistoryArgs
+  redelegations?: GQLQueryGqlRedelegationsArgs
+  unbondings?: GQLQueryGqlUnbondingsArgs
+  validators?: GQLStakingGqlValidatorsArgs
+}
+
+export interface GqlOutStaking {
+  staking?: GQLQuery["staking"]
+}
+
+export type GQLStakingFields = DeepPartial<{
```

Comment:
Can be removed in favor of `GQLStaking`

Reply:
DeepPartial<GQLStaking> *

Reply:
https://github.com/NibiruChain/ts-sdk/pull/347#discussion_r1580082223

Reply:
https://github.com/NibiruChain/ts-sdk/pull/347#discussion_r1581706548